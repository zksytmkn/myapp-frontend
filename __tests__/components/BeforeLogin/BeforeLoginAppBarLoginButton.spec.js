import { shallowMount } from '@vue/test-utils';
import BeforeLoginAppBarLoginButton from '@/components/BeforeLogin/BeforeLoginAppBarLoginButton.vue';

describe('BeforeLoginAppBarLoginButton.vue', () => {
  let wrapper;

  beforeEach(() => {
    // $my オブジェクトをモック
    const $my = {
      pageTitle: jest.fn().mockReturnValue('Login')
    };

    // コンポーネントをマウントする
    wrapper = shallowMount(BeforeLoginAppBarLoginButton, {
      mocks: {
        $my
      }
    });
  });

  it('renders correctly', () => {
    // コンポーネントが正しくレンダリングされているか確認
    expect(wrapper.element).toMatchSnapshot();
  });

  it('has the correct default properties', () => {
    // デフォルトのプロパティが正しいか確認
    const button = wrapper.find('v-btn');
    expect(button.classes()).toContain('ml-2');
    expect(button.classes()).toContain('font-weight-bold');
    expect(button.attributes('color')).toBe('black');
    expect(button.attributes('text')).toBe(''); // 'text' property exists
  });

  it('displays the correct page title', () => {
    // ボタン内のテキストが正しいか確認
    const buttonText = wrapper.find('v-btn').text();
    expect(buttonText).toContain('Login');
  });

  it('navigates to the correct page on click', () => {
    // クリックしたときに正しいページに遷移するか確認
    const button = wrapper.find('v-btn');
    expect(button.attributes('to')).toBe('/login');
  });
});
