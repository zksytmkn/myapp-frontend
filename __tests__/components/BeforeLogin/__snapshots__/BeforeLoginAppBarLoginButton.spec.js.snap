// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BeforeLoginAppBarLoginButton.vue renders correctly 1`] = `
<v-btn
  class="ml-2 font-weight-bold"
  color="black"
  text=""
  to="/login"
>
  
  Login

</v-btn>
`;
